src/world/map.c

```diff
@@ -3014,191 +3014,240 @@ void game_command_place_scenery(int* eax, int* ebx, int* ecx, int* edx, int* esi
-									bp &= 1;
-									bl = 0xA;
-									bl >>= bp;
-								}
-							}
-						}else{
-							bp ^= 2;
-							bp += rotation;
-							bp &= 3;
-							bl = 0x33;
-							bl = rol8(bl, bp);
-							bl &= 0xF;
-						}
-						if(z == 0){
-							bl |= 0xF0;
-						}
 
-						if(gCheatsDisableClearanceChecks || map_can_construct_with_clear_at(x, y, zLow, zHigh, &map_place_scenery_clear_func, bl, flags, &clearCost)){
-							gSceneryGroundFlags = gMapGroundFlags & (ELEMENT_IS_1 | ELEMENT_IS_UNDERGROUND);
-							if(flags & GAME_COMMAND_FLAG_APPLY){
-								if (gGameCommandNestLevel == 1 && !(flags & GAME_COMMAND_FLAG_GHOST)) {
-									rct_xyz16 coord;
-									coord.x = x + 16;
-									coord.y = y + 16;
-									coord.z = map_element_height(coord.x, coord.y);
-									network_set_player_last_action_coord(network_get_player_index(game_command_playerid), coord);
-								}
-								int collisionQuadrants = (bl & 0xf);
-								rct_map_element* new_map_element = map_element_insert(x / 32, y / 32, zLow, collisionQuadrants);
-								assert(new_map_element != NULL);
-								gSceneryMapElement = new_map_element;
-								uint8 type = quadrant << 6;
-								type |= MAP_ELEMENT_TYPE_SCENERY;
-								type |= rotation;
-								new_map_element->type = type;
-								new_map_element->properties.scenery.type = scenery_type;
-								new_map_element->properties.scenery.age = 0;
-								new_map_element->properties.scenery.colour_1 = colour1;
-								new_map_element->properties.scenery.colour_2 = colour2;
-								new_map_element->clearance_height = new_map_element->base_height + ((scenery_entry->small_scenery.height + 7) / 8);
-								if(z != 0){
-									new_map_element->properties.scenery.colour_1 |= 0x20;
-								}
-								if(flags & 0x40){
-									new_map_element->flags |= 0x10;
-								}
-								map_invalidate_tile_full(x, y);
-								if(scenery_entry->small_scenery.flags & SMALL_SCENERY_FLAG_ANIMATED){
-									map_animation_create(2, x, y, new_map_element->base_height);
-								}
-							}
-							*ebx = (scenery_entry->small_scenery.price * 10) + clearCost;
-							if(gParkFlags & PARK_FLAGS_NO_MONEY){
-								*ebx = 0;
-							}
-							return;
-						}
-					}else{
-						if(F64F1D == 0){
-							if((map_element->properties.surface.terrain & 0x1F) || (map_element->base_height * 8) != F64EC8){
-								gGameCommandErrorText = STR_LEVEL_LAND_REQUIRED;
-							}else{
-								goto l_6E0B78;
-							}
-						}else{
-							gGameCommandErrorText = STR_CAN_ONLY_BUILD_THIS_ON_LAND;
-						}
-					}
-				}else{
-					gGameCommandErrorText = STR_LEVEL_LAND_REQUIRED;
-				}
+	if (game_is_paused() && !gCheatsBuildInPauseMode) {
+		gGameCommandErrorText = STR_CONSTRUCTION_NOT_POSSIBLE_WHILE_GAME_IS_PAUSED;
+		*ebx = MONEY32_UNDEFINED;
+		return;
+	}
+
+	if (!map_check_free_elements_and_reorganise(1)) {
+		*ebx = MONEY32_UNDEFINED;
+		return;
+	}
+
+	if (!(byte_9D8150 & 1) && (x > gMapSizeMaxXY || y > gMapSizeMaxXY)) {
```

Comment:
I can't work out why this is used. As far as I can tell it's an override for when the track window but not sure why you would want to place outside of the x y limits during this but not on any other scenery